tests/queries/tests.py

```diff
@@ -2993,11 +3003,6 @@ def test_empty_nodes(self):
 
 
 class QuerySetExceptionTests(TestCase):
-    def test_iter_exceptions(self):
```

Comment:
It looks like this is fixing another issue -- probably the test should be updated and moved to `tests/defer`. 

Could you check https://code.djangoproject.com/ticket/23051 and https://code.djangoproject.com/ticket/24612. If those tickets don't describe the issue that's fixed, please open a new ticket. If you can send the fix as a separate pull request, that would be great.

Reply:
This one is the last one remaining.
At first skim, existing tickets doesn't cover those changes.
I will probably open a new ticket.

Reply:
I reverted the change and decided to document that
`FilteredRelation` and `only()` are not supported together.
I also changed the test to reflect this decision.

Because the test `QuerySetExceptionTests.test_iter_exceptions` exists to prevent user from doing something not supported.
We shouldn’t change that behaviour.

```python
qs = ExtraInfo.objects.only('author')
```
This snippet shouldn't be allowed today, because from `ExtraInfo` point of view, `.only('author')` match  `author_set` on the instance, which is not desirable I think.